### ISSUE TITLE
Error reported at closing

### ISSUE THREAD

**Issue by inspectionsbybob** [meerk40t/meerk40t#315]
System said to send this to you, so I am...

Saw this when I was closing down for the evening to work on some other things....

![error2](https://user-images.githubusercontent.com/22258531/105618334-ed7f3c80-5db3-11eb-8c97-4f4ed47a5070.png)

**Comment by tatarize** [comment]
I'm guessing that's 0.6.17 beta. Or one of them.

There's a standing issue there with spinners in gtk version doing a strange ordering with the save. It's a kinda trivial bug but a headscratcher on fixing it. 

Mac and Linux users if they have the spinner focused and they close the window. The setting they set in the spinner does not save. So I tried a bunch of stuff to fix that. And then just corrected any crashing it would do. It still does not save. I got rid of spinners for important areas since they have *other* bugs too. Namely the SpinCtrlDouble in GTK only does integers, which is absolute death in a number of places.

I restored the anti-crash stuff after those betas and ran into another windows bug that I thought was maybe caused by that. Loading up navigation and then closing the main window to kill the sub-windows works, but it doesn't shutdown gracefully and often crashes after closing.

It's a known error. I just don't know how to fix it, and might just restore the failed stuff. The idea of maintaining the device reference after window close seems to have opened up a can of worms. And it's fixed in 0.7.0 so it's a bit of a pain in the ass overall.
